astropy/io/ascii/tdat.py

```diff
@@ -0,0 +1,853 @@
+        # In case a user puts a keyword directly in meta, instead of meta.keywords
+        for key in [
+            key.lower()
+            for key in self.table_meta.keys()
+            if key.lower() not in meta_keys
+        ]:
+            keywords[key] = self.table_meta.get(key)
+
+        indices = [col.info.name for col in self.cols if col.info.indices != []]
+
+        if "table_name" in keywords:
+            if len(keywords["table_name"]) > 20:
+                warn(
+                    "'table_name' is too long, truncating to 20 characters",
+                    TdatFormatWarning,
+                )
+            lines.append(f"table_name = {keywords.pop('table_name')[:20]}")
+        else:
+            warn(
+                "'table_name' must be specified\n"  # noqa: ISC003
+                + f"{_STD_MSG}\n"
+                + "This should be specified in the Table.meta.\n"
+                + "default value of 'astropy_table' being assigned.",
+                TdatFormatWarning,
+            )
+            lines.append("table_name = astropy_table")
+
+        # loop through optional table keywords
+        table_keywords = [
+            "table_description",
+            "table_document_url",
+            "table_security",
+        ]
+        table_keywords = [kw for kw in table_keywords if kw in keywords.keys()]
+        for kw in ["record_delimiter", "field_delimiter"]:
+            val = keywords.pop(kw, None)
+            if val is not None:
+                warn(TdatFormatWarning(f"Skipping deprecated keyword: {kw}."))
+
+        for key in table_keywords:
+            if key == "table_description":
+                if len(keywords[key]) > 80:
+                    warn(
+                        "'table_description' is too long, truncating to 80 characters",
+                        TdatFormatWarning,
+                    )
+                new_desc = keywords.pop(key)
+                new_desc = new_desc[:80]
+                lines.append(f"{key} = {new_desc}")
+            elif key == "table_security":
+                if keywords.get(key).lower() not in ("public", "private"):
+                    warn(
+                        "Value for table_security not recognized, should be public|private.",
+                        TdatFormatWarning,
+                    )
+                lines.append(f"{key} = {keywords.pop(key)}")
+            elif key in keywords:
+                lines.append(f"{key} = {keywords.pop(key)}")
+
+        # add table columns as fields
+        lines.append("#")
+        lines.append("# Table Parameters")
+        lines.append("#")
+        for col in self.cols:
+            if str(col.info.dtype) in self._dtype_dict_out:
+                ctype = self._dtype_dict_out[str(col.info.dtype)]
+            elif "int" in str(col.info.dtype):
+                ctype = "int4"
+            elif "float" in str(col.info.dtype):
+                ctype = "float8"
+            elif "<U" in str(col.info.dtype):
```

Comment:
Is little-endian mandated by the format, or could we also allow `>U` (or `=U`?) as dtype here? This is apparently the cause of #17997.

Reply:
The format is ASCII, so it should be unaffected by endianness.

Reply:
That's more or less what I expected, too. So why specifically filtering for little-endian types here? Note that I haven't delved far into the context yet.